### DISCUSSION TITLE
Async Event Aggregator

### DISCUSSION THREAD

**Discussion by alexhelms** [canton7/Stylet#238] [Ideas]
I'm trying out Stylet and so far I'm definitely enjoying using it! I started to play with the event aggregator and discovered that it is entirely synchronous.

My reasoning for adding async APIs is in my toy project I am using MediatR notifications and the problem with MediatR notifications are the handlers are transient and are created per notification which obviously wont work for view models acting as notification handlers. My solution to this is to have essentially a translator to handle the INotification and send out the appropriate Stylet event used for the view models. MediatR handlers are generally async and it would be nice to keep everything async when converting between MediatR notifications and event aggregator events.

Would there be any interest in adding async methods to the event aggregator API?

**Comment by canton7** [comment]
Mind jumping into #189?

**Reply by alexhelms** [reply]
I promise I searched! 😄 I'll move my comments over there, thanks.

**Reply by canton7** [reply]
No worries! Step 1 I think is if you have answers to the questions I asked there.
